dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,546 @@
+
+    """
+    def __init__(self, model, params, max_iter=81, eta=3, asynchronous=None,
+                 random_state=42, scoring=None, test_size=0.15):
+        self.model = model
+        self.params = params
+        self.max_iter = max_iter
+        self.eta = eta
+        self.test_size = test_size
+        self.random_state = random_state
+        if asynchronous is None:
+            asynchronous = True
+        self.asynchronous = asynchronous
+        self._best_score = -np.inf
+
+        super(HyperbandCV, self).__init__(model, scoring=scoring)
+
+    def fit(self, X, y, **fit_params):
+        """Find the best parameters for a particular model
+
+        Arguments
+        ---------
+        X, y : two dask.arrays or np.ndarrays
+            Input to ``fit``. X.ndim == 2 and y.ndim == 1. If dask arrays are
+            passed, each ``partial_fit`` call will be over one chunk of the
+            array.
+        fit_params : dict
+            Arguments to pass to ``model.partial_fit``
+        """
+        return default_client().sync(self._fit, X, y, **fit_params)
+
+    @gen.coroutine
+    def _fit(self, X, y, **fit_params):
+        if isinstance(X, np.ndarray):
+            X = da.from_array(X, chunks=X.shape)
+        if isinstance(y, np.ndarray):
+            y = da.from_array(y, chunks=y.shape)
+
+        # We always want a concrete scorer, so return_dask_score=False
+        # We want this because we're always scoring NumPy arrays
+        self.scorer_ = check_scoring(self.model, scoring=self.scoring)
+        r = yield _hyperband(self.model, self.params, X, y,
+                             max_iter=self.max_iter,
+                             fit_params=fit_params, eta=self.eta,
+                             random_state=self.random_state,
+                             test_size=self.test_size,
+                             scorer=self.scorer_,
+                             asynchronous=self.asynchronous,
+                             callback=self._logging_callback)
+        params, model_meta_futures, history, meta = r
+
+        self.meta_ = meta
+        self.history_ = history
+
+        cv_res, best_idx = _get_cv_results(meta, params)
+        best_id = cv_res['model_id'][best_idx]
+
+        best_model_and_meta = yield model_meta_futures[best_id]
+        self.best_estimator_ = best_model_and_meta[0]
+
+        self.cv_results_ = cv_res
+        self.best_index_ = best_idx
+        self.n_splits_ = 1  # TODO: increase this! It's hard-coded right now
+        self.multimetric_ = False
+
+        return self
+
+    def _logging_callback(self, result):
+        if result['mean_test_score'] > self._best_score:
+            msg = '[CV] new best validation score of {} found with params {}'
+            logger.info(msg.format(result['mean_test_score'],
+                                   result['params']))
+            self._best_score = result['mean_test_score']
+
+    def fit_metadata(self, meta=None):
```

Comment:
Should this method be made private?

Reply:
I don't think so. I mention it in the docs on how to choose `max_iter` and how to chunk dask arrays.

Reply:
I know I would appreciate this function: it would determine how long computation will take, and the memory requirements.